src/app/users-page/author/add-authors/add-authors.component.spec.ts

```diff
@@ -1,28 +1,43 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { AddAuthorsComponent } from './add-authors.component';
 import { ReactiveFormsModule, FormBuilder, FormsModule } from '@angular/forms';
-import { By } from '@angular/platform-browser';
+import { Author } from 'src/app/models/author';
+import { AuthorService } from '../../../services/author.service';
+import { Observable } from 'rxjs';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { DebugElement } from '@angular/core';
-import { HttpClientModule } from '@angular/common/http';
+import { BrowserModule, By } from '@angular/platform-browser';
+
+class MockService {
+  addAuthor(): Observable<Author[]> {
+    return {} as Observable<Author[]>;
+  }
+}
 
 describe('AddAuthorsComponent', () => {
   let authorComponent: AddAuthorsComponent;
   let fixture: ComponentFixture<AddAuthorsComponent>;
-  let de: DebugElement;
-  let el: HTMLElement;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [AddAuthorsComponent],
-      imports: [ReactiveFormsModule, FormsModule, HttpClientModule],
+      imports: [
+        BrowserModule,
+        ReactiveFormsModule,
+        FormsModule,
+        HttpClientTestingModule,
+      ],
+      providers: [
+        {
+          AuthorService,
+          useValue: MockService,
+        },
+      ],
     }).compileComponents();
 
     fixture = TestBed.createComponent(AddAuthorsComponent);
     authorComponent = fixture.componentInstance;
     authorComponent.ngOnInit();
```

Comment:
The **ngOnInit** method of this component makes a call to **getAllAuthors** which in turn fetches authors from the backend. Consider also adding this method to your **mockService** above to prevent this network call.